src/CodeGenHelpers/ConstructorBuilder.cs

```diff
@@ -21,12 +21,36 @@ internal ConstructorBuilder(Accessibility? accessModifier, ClassBuilder classBui
             Class = classBuilder;
         }
 
+        public DocumentationComment XmlDoc { get; } = new DocumentationComment();
+
         public Accessibility? AccessModifier { get; }
 
         public ClassBuilder Class { get; }
 
         internal int Parameters => _parameters.Count;
 
+        public ConstructorBuilder WithSummary(string summary)
+        {
+            XmlDoc.Summary = summary;
+            return this;
+        }
+
+        public ConstructorBuilder WithInheritDoc(bool inherit = true, string from = null)
+        {
+            XmlDoc.InheritDoc = inherit;
+            XmlDoc.InheritFrom = from;
+            return this;
+        }
+
+        public ConstructorBuilder WithParameterDoc(string paramName, string documentation)
+        {
+            // The reason why I don't check if the parameter exists,
+            // is that maybe the user wants to add the parameter later themselves
+            // and an extra xmldoc won't really cause issue.
+            XmlDoc.ParameterDoc[paramName] = documentation;
```

Comment:
this should probably have a check when we call Write so that we remove parameters that don't exist